providers/sugarcrm.go

```diff
@@ -0,0 +1,29 @@
+package providers
+
+const SugarCRM Provider = "sugarCRM"
+
+func init() {
+	// 2-legged auth
+	SetInfo(SugarCRM, ProviderInfo{
+		AuthType: Oauth2,
+		BaseURL:  "{{.workspace}}",
+		Oauth2Opts: &Oauth2Opts{
+			GrantType:                 Password,
+			TokenURL:                  "{{.workspace}}/rest/{{.restVersion}}/oauth2/token",
```

Comment:
let's hard code in a restVersion right now, we don't support a way to get the restVersion from the builder currently, so this won't work with our existing system.

Reply:
I updated the base URL to match the version and restrict to rest usage. https://github.com/amp-labs/connectors/pull/602/commits/c8a11f40ac59c98aa88fcf2442aabf895e27a1e4